trio/_core/_run.py

```diff
@@ -1478,14 +1478,20 @@ class RunStatistics:
 # So this object can reference Runner, but Runner can't reference it. The only
 # references to it are the "in flight" callback chain on the host loop /
 # worker thread.
+
+
+def _value_factory() -> Value[None]:
+    return Value(None)
+
+
 @attr.s(eq=False, hash=False, slots=True)
 class GuestState:
     runner: Runner = attr.ib()
     run_sync_soon_threadsafe: Callable[[Callable[[], object]], object] = attr.ib()
     run_sync_soon_not_threadsafe: Callable[[Callable[[], object]], object] = attr.ib()
     done_callback: Callable[[Outcome[Any]], object] = attr.ib()
     unrolled_run_gen: Generator[float, EventResult, None] = attr.ib()
-    _value_factory: Callable[[], Value[Any]] = lambda: Value(None)
+
```

Comment:
nit: I'd remove this newline.

Reply:
... Why was this changed anyways? I don't really see something based on the lines around here.

Reply:
Basically because one of the ruff rules is against using lamdas, and really why was this defined as an attribute of GuestState anyways?

Removed newline in 561453ea

Reply:
Yeah I'm not sure why it's an attribute of GuestState. I'm not sure why it's not inlined in the `factory=` bit either...

Reply:
Made factory inlined with 8f88afc